tikv-client/src/main/java/com/pingcap/tikv/ClientSession.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2022 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pingcap.tikv;
+
+import com.pingcap.tikv.catalog.Catalog;
+import com.pingcap.tikv.util.ConvertUpstreamUtils;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import lombok.Getter;
+import org.tikv.common.TiSession;
+import org.tikv.common.event.CacheInvalidateEvent;
+import org.tikv.common.meta.TiTimestamp;
+
+@Getter
+public class ClientSession implements AutoCloseable {
+
+  private static final Map<String, ClientSession> sessionCachedMap = new HashMap<>();
+  private final TiConfiguration conf;
+  private final TiSession tiKVSession;
+  private final Catalog catalog;
```

Comment:
why can't we use the upstream tiKVSession's catalog and snapshot?

Reply:
Catalog and Snapshot use classes in the com.pingcap.tikv.meta package. This package involves private information maintained by tispark. Upstream replacement cannot be used.